src/sparsezoo/api/query_parser.py

```diff
@@ -93,12 +93,64 @@ def _parse_arguments(self) -> None:
 
     def _parse_fields(self) -> None:
         fields = self.fields or DEFAULT_FIELDS.get(self.operation_body)
-
+        if isinstance(fields, List):
+            self.fields = self.parse_list_fields_to_string(fields)
+        elif isinstance(fields, Dict):
+            """
+            fields = {
+                "model_id": None,
+                "benchmarks": {
+                    benchmark_result_id: None,
+                    whatever: {
+                        ...
+                    }
+                }
+            }
+            """
+            self.fields = self.parse_dict_fields_to_string(fields)
+
+    def parse_list_fields_to_string(self, fields: List[str]) -> str:
         parsed_fields = ""
         for field in fields:
             camel_case_field = to_camel_case(field)
             parsed_fields += f"{camel_case_field} "
-        self.fields = parsed_fields
+            if camel_case_field in DEFAULT_FIELDS:
+                parsed_fields += (
+                    "{ "
+                    + self.parse_list_fields_to_string(
+                        DEFAULT_FIELDS.get(camel_case_field)
+                    )
+                    + "} "
+                )
```

Comment:
I think I would pull out a local variable here:
```suggestion
                fields_for_field = self.parse_list_fields_to_string(
                    DEFAULT_FIELDS.get(camel_case_field)
                )
                parsed_fields += f"{{ {fields_for_field}}} "
```

I tried to match your spacing around the braces. Is that intentional spacing?